docs/resources/aws_kms_key.md.erb

```diff
@@ -0,0 +1,171 @@
+---
+title: About the aws_kms_key Resource
+---
+
+# aws\_kms\_key
+
+Use the `aws_kms_key` InSpec audit resource to test properties of a single AWS KMS Key.
+
+Use aws_kms_key to verify the properties of a single key. Use aws_kms_keys to verify the properties of all or a group of keys.
+
+AWS Key Management Service (AWS KMS) is a managed service that makes it easy for you to create and control the encryption keys used to encrypt your data. AWS KMS lets you create master keys that can never be exported from the service and which can be used to encrypt and decrypt data based on policies you define.
+
+Each AWS KMS Key is uniquely identified by its key_id or arn.
+
+<br>
+
+## Syntax
+
+An aws_kms_key resource block identifies a key by key_arn or the key id.
+
+    # Find a kms key by arn
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      it { should exist }
+    end
+    
+    # Find a kms key by just the id
+    describe aws_kms_key('4321dcba-21io-23de-85he-ab0987654321') do
+      it { should exist }
+    end
+
+    # Hash syntax for key arn
+    describe aws_kms_key(key_arn: 'arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      it { should exist }
+    end
+
+<br>
+
+## Examples
+
+The following examples show how to use this InSpec audit resource.
+
+### Test that the specified key does exist
+
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      it { should exist }
+    end
+
+### Test that the specified key is enabled
+
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      it { should be_enabled }
+    end
+
+### Test that the specified key is rotation enabled
+
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      it { should have_rotation_enabled }
+    end
+
+<br>
+
+## Properties
+
+### key\_id
+
+The globally unique identifier for the key.
+
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      its('key_id') { should cmp 'arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321' }
```

Comment:
I thought the key ID was just the GUID portion - `4321dcba-21io-23de-85he-ab0987654321`.  It seems confusing that here we're testing against an ARN-format expected value.